addons/web/static/src/legacy/js/widgets/colorpicker.js

```diff
@@ -41,6 +41,7 @@ var ColorpickerWidget = Widget.extend({
 
         // Needs to be bound on document to work in all possible cases.
         const $document = $(
+            parent.options.ownerDocument ||
```

Comment:
For that I was thinking that an alternative could be to register the event listeners once the ColorpickWidget element is inserted in the DOM.
Doing that, we would not need anything from the parent, and we are sure to always have the appropriate document. 
Unfortunately the MutationObserver does not seem to monitor parents mutations